lib/src/kbasesearchengine/main/IndexerWorker.java

```diff
@@ -414,35 +414,40 @@ private EventHandler getEventHandler(final String storageCode)
 
     private void processEvent(final StatusEvent ev)
             throws IndexingException, InterruptedException, RetriableIndexingException {
+
+        // update event to reflect the latest state of object for which the specified StatusEvent is an event for
+        EventHandler handler = getEventHandler(ev.getStorageCode());
```

Comment:
I still feel like a couple integration tests around deleted objects and renamed/republished objects would be good, but you're the bossman now so your call.

Reply:
Added a couple for deletion and rename as a sanity check.